CalibTracker/SiStripHitResolution/plugins/SiStripHitResolFromCalibTree.cc

```diff
@@ -0,0 +1,1730 @@
+  bool _useCM;
+  bool _showEndcapSides;
+  bool _showRings;
+  bool _showTOB6TEC9;
+  bool _showOnlyGoodModules;
+  float _tkMapMin;
+  float _effPlotMin;
+  TString _title;
+
+  edm::ESGetToken<TrackerGeometry, TrackerDigiGeometryRecord> _tkGeomToken;
+  edm::ESGetToken<TrackerTopology, TrackerTopologyRcd> _tTopoToken;
+
+  unsigned int nTEClayers;
+
+  TH1F* bxHisto;
+  TH1F* instLumiHisto;
+  TH1F* PUHisto;
+
+  // for association of informations of the hitEff tree and the event infos tree
+  map<pair<unsigned int, unsigned int>, array<double, 3> > eventInfos;
+
+  vector<hit> hits[23];
+  vector<TH2F*> HotColdMaps;
+  map<unsigned int, pair<unsigned int, unsigned int> > modCounter[23];
+  TrackerMap* tkmap;
+  TrackerMap* tkmapbad;
+  TrackerMap* tkmapeff;
+  TrackerMap* tkmapnum;
+  TrackerMap* tkmapden;
+  long layerfound[23];
+  long layertotal[23];
+  map<unsigned int, vector<int> > layerfound_perBx;
+  map<unsigned int, vector<int> > layertotal_perBx;
+  vector<TH1F*> layerfound_vsLumi;
+  vector<TH1F*> layertotal_vsLumi;
+  vector<TH1F*> layerfound_vsPU;
+  vector<TH1F*> layertotal_vsPU;
+  vector<TH1F*> layerfound_vsCM;
+  vector<TH1F*> layertotal_vsCM;
+  int goodlayertotal[35];
+  int goodlayerfound[35];
+  int alllayertotal[35];
+  int alllayerfound[35];
+  map<unsigned int, double> BadModules;
+};
+
+SiStripHitResolFromCalibTree::SiStripHitResolFromCalibTree(const edm::ParameterSet& conf)
+    : ConditionDBWriter<SiStripBadStrip>(conf), FileInPath_("CalibTracker/SiStripCommon/data/SiStripDetInfo.dat") {
+  usesResource(TFileService::kSharedResource);
+  CalibTreeFilenames = conf.getUntrackedParameter<vector<std::string> >("CalibTreeFilenames");
+  threshold = conf.getParameter<double>("Threshold");
+  nModsMin = conf.getParameter<int>("nModsMin");
+  doSummary = conf.getParameter<int>("doSummary");
+  _badModulesFile = conf.getUntrackedParameter<std::string>("BadModulesFile", "");
+  _autoIneffModTagging = conf.getUntrackedParameter<bool>("AutoIneffModTagging", false);
+  _clusterMatchingMethod = conf.getUntrackedParameter<int>("ClusterMatchingMethod", 0);
+  _ResXSig = conf.getUntrackedParameter<double>("ResXSig", -1);
+  _clusterTrajDist = conf.getUntrackedParameter<double>("ClusterTrajDist", 64.0);
+  _stripsApvEdge = conf.getUntrackedParameter<double>("StripsApvEdge", 10.0);
+  _useOnlyHighPurityTracks = conf.getUntrackedParameter<bool>("UseOnlyHighPurityTracks", true);
+  _bunchx = conf.getUntrackedParameter<int>("BunchCrossing", 0);
+  _spaceBetweenTrains = conf.getUntrackedParameter<int>("SpaceBetweenTrains", 25);
+  _useCM = conf.getUntrackedParameter<bool>("UseCommonMode", false);
+  _showEndcapSides = conf.getUntrackedParameter<bool>("ShowEndcapSides", true);
+  _showRings = conf.getUntrackedParameter<bool>("ShowRings", false);
+  _showTOB6TEC9 = conf.getUntrackedParameter<bool>("ShowTOB6TEC9", false);
+  _showOnlyGoodModules = conf.getUntrackedParameter<bool>("ShowOnlyGoodModules", false);
```

Comment:
this block has `_string` or `_String` naming structure, it would be better to have them all as `string_`

Reply:
done.